CohortsScheduling/src/main/java/controller/ScheduleController.java

```diff
@@ -1,24 +1,49 @@
 package controller;
 
+import dataModels.StartRequest;
+import runnable.CohortRunnableFactory;
+import runnable.ScheduleRunnable;
+
 public class ScheduleController {
 
-	public static String start() {
-		//validates info and starts a thread, returns success or fail if exception thrown
-		return null;
-		//starts scheduling
+	private static boolean cohortCalcRunning;
+	private static ScheduleRunnable currentScheduler;
+	private static Thread optThread;
+	public static void init() {
+		cohortCalcRunning = false;
+		currentScheduler = null;
 	}
 	
-	public static String status() {
-		return null;
-
-		//called by service, checks if a thread is running
+	public static String start(StartRequest request) {
+		if(cohortCalcRunning) {
+			return "Already running";
+		}
+		
```

Comment:
Do parsing and validation here. Pass the runnable factory an array of CohortSolution